### PR TITLE
[Behat] Cleaning Behat checkout pages

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php [hunk 1]
```diff
@@ -90,12 +81,7 @@ public function checkInvalidCredentialsValidation()
         return $this->checkValidationMessageFor('login_password', 'Invalid credentials.');
     }
 
-    /**
-     * {@inheritdoc}
-     *
-     * @throws ElementNotFoundException
-     */
-    public function checkValidationMessageFor($element, $message)
+    public function checkValidationMessageFor($element, $message): bool
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php]: What about adding typehints?

File: src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php [hunk 2]
```diff
@@ -110,71 +96,47 @@ public function checkValidationMessageFor($element, $message)
...
+    public function specifyShippingAddress(AddressInterface $shippingAddress): void
     {
         $this->specifyAddress($shippingAddress, self::TYPE_SHIPPING);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function selectShippingAddressProvince($province)
+    public function selectShippingAddressProvince($province): void
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php]: Missing typehint here and in other changed methods

File: src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php [hunk 3]
```diff
@@ -90,12 +81,7 @@ public function checkInvalidCredentialsValidation()
         return $this->checkValidationMessageFor('login_password', 'Invalid credentials.');
     }
 
-    /**
-     * {@inheritdoc}
-     *
-     * @throws ElementNotFoundException
-     */
-    public function checkValidationMessageFor($element, $message)
+    public function checkValidationMessageFor(string $element, string $message): bool
```
Comment (Tomanhez) [inline src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php]: in line 88, you should change validation error from class do data-attributes like this https://github.com/Sylius/Sylius/pull/10883/files#diff-22c371129177dc765960e9e7aa87a410R31

File: src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php [hunk 4]
```diff
@@ -73,10 +67,7 @@ public function chooseDifferentBillingAddress()
...
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function checkInvalidCredentialsValidation()
+    public function checkInvalidCredentialsValidation(): bool
     {
         $this->getElement('login_password')->waitFor(5, function () {
             $validationElement = $this->getElement('login_password')->getParent()->find('css', '.red.label');
```
Comment (Tomanhez) [inline src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php]:
```suggestion
            $validationElement = $this->getElement('login_password')->getParent()->find('css', '.red.label');
```

remove `.red.label` and add data-attributes

File: src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php [hunk 5]
```diff
@@ -211,62 +167,44 @@ public function getItemSubtotal($itemName)
         return $subtotalTable->find('css', sprintf('#sylius-item-%s-subtotal', $itemSlug))->getText();
```
Comment (Tomanhez) [inline src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php]:
```suggestion
        return $subtotalTable->find('css', sprintf('data-test-sylius-item-%s-subtotal', $itemSlug))->getText();
```
?

File: src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php [hunk 6]
```diff
@@ -211,62 +167,44 @@ public function getItemSubtotal($itemName)
...
+    public function hasBillingAddressInput(): bool
     {
         return $this->waitForElement(5, 'billing_province');
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function selectShippingAddressFromAddressBook(AddressInterface $address)
+    public function selectShippingAddressFromAddressBook(AddressInterface $address): void
```
Comment (Tomanhez) [inline src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php]: in line 214 : 
```
return $addressBookSelect->find('css', sprintf('.item[data-id="%s"]', $address->getId()));
```

change like 
```
return $addressBookSelect->find('css', sprintf('[data-test-item-id="%s"]', $address->getId()));
```

File: src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php [hunk 7]
```diff
@@ -285,10 +223,7 @@ public function selectShippingAddressFromAddressBook(AddressInterface $address)
         JQueryHelper::waitForFormToStopLoading($this->getDocument());
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function selectBillingAddressFromAddressBook(AddressInterface $address)
+    public function selectBillingAddressFromAddressBook(AddressInterface $address): void
```
Comment (Tomanhez) [inline src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php]: and here 233 and 237 replace `.item[data-id="%s"]` for example: `[data-test-item-id="%s"]`

File: src/Sylius/Behat/Page/Shop/Checkout/CompletePageInterface.php [hunk 8]
```diff
@@ -20,71 +20,29 @@
...
+
+    public function getBaseCurrencyOrderTotal(): string;
+
+    public function hasShippingMethod(ShippingMethodInterface $shippingMethod): bool;
+
+    public function addNotes(string $notes): void;
+
+    public function hasPromotionTotal(string $promotionTotal): bool;
+
+    public function hasPromotion(string $promotionName): bool;
```
Comment (Tomanhez) [inline src/Sylius/Behat/Page/Shop/Checkout/CompletePageInterface.php]: maybe you should put it in alphabetical order?

Comment (AdamKasp) [reply src/Sylius/Behat/Page/Shop/Checkout/CompletePageInterface.php]: IMO it is unnecessary, there is no coding standard telling about that

File: src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php (hunk 3 above)
Comment (AdamKasp) [reply src/Sylius/Behat/Page/Shop/Checkout/AddressPage.php]: here and bottom: on this PR I only added type hints and deleted unnecessary doc blocks

**Comment by lchrusciel** [comment]
Thank you, Adam! :tada:

**Pull request merged**